### PR TITLE
Fixed tests for src/setup/setImageUploadSize.ts

### LINKED ISSUE #1924: Test: src/setup/setImageUploadSize.ts
The Talawa-API code base needs to be 100% reliable. This means we need to have 100% test code coverage.

1. Tests need to be written for file `src/setup/setImageUploadSize.ts`
1. We will need the API to be refactored for all methods, classes and/or functions found in this file for testing to be correctly executed.
1. When complete, all all methods, classes and/or functions in the refactored file will need to be tested. 
1. These tests must be placed in a single file with the same path as the tested file, but under the `tests/` with a `.spec.ts` suffix.
1. You may need to create the appropriate directory structure to do this.

### IMPORTANT:

Please refer to the parent issue on how to implement these tests correctly:

- https://github.com/PalisadoesFoundation/talawa-api/issues/490

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base. This will be clearly visible when you submit your PR.
- [The current code coverage for the file can be found by visting this page](https://app.codecov.io/gh/PalisadoesFoundation/talawa-api?displayType=list). Login using your GitHub credentials.
- Create your own `codecov.io` to help with testing.
- The PR you create will show a report for the code coverage for the file you have added. You can also use that as a guide.

### Other

Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue:

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aarishshahmohsin**
**What kind of change does this PR introduce?**
Fixed tests for src/setup/setImageUploadSize.ts increasing coverage.

**Issue Number:**
Fixes #1924 

**Did you add tests for your changes?**
NA

**Snapshots/Videos:**
NA

**If relevant, did you update the documentation?**
NA

**Summary**
Fixed tests for src/setup/setImageUploadSize.ts increasing coverage.

**Does this PR introduce a breaking change?**
No

**Other information**
NA

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by Olatade** [comment]
@aarishshahmohsin can you fix the failing tests please

**Comment by Cioppolo14** [comment]
@SiddheshKukade @EshaanAgg Can you review this PR?

**Comment by palisadoes** [comment]
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by palisadoes** [comment]
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by palisadoes** [comment]
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

File: tests/setup/setImageUploadSize.spec.ts [hunk 1]
```diff
@@ -1,63 +1,92 @@
...
 import {
   setImageUploadSize,
   validateImageFileSize,
 } from "../../src/setup/setImageUploadSize";
+import fs from "fs";
+import dotenv from "dotenv";
 import { MAXIMUM_IMAGE_SIZE_LIMIT_KB } from "../../src/constants";
 
-/*
-  Test Case 1:
```
Comment (palisadoes) [inline tests/setup/setImageUploadSize.spec.ts:11]: Why have you removed these clarifying comments?

File: tests/setup/setImageUploadSize.spec.ts [hunk 2]
```diff
@@ -1,63 +1,92 @@
...
-  Expected Behavior: When setImageUploadSize function is called with a size value, it should update the .env_test file with the provided size in kilobytes.
+describe("setImageUploadSize", () => {
+  const originalEnv = { ...process.env };
+
+  beforeEach(() => {
+    process.env = { ...originalEnv };
+  });
 
-  Test Case 2:
-  Description: setImageUploadSize truncates size if exceeding the maximum limit.
```
Comment (palisadoes) [inline tests/setup/setImageUploadSize.spec.ts:16]: Why have you removed these clarifying comments?

File: tests/setup/setImageUploadSize.spec.ts [hunk 3]
```diff
@@ -1,63 +1,92 @@
...
-  Expected Behavior: When setImageUploadSize function is called with a size value exceeding the maximum limit, it should truncate the size to the maximum limit defined by MAXIMUM_IMAGE_SIZE_LIMIT_KB constant and update the .env_test file accordingly.
+  afterEach(() => {
+    process.env = originalEnv;
+    vi.restoreAllMocks();
+  });
 
-  Test Case 3:
-  Description: The validateImageFileSize function returns true for a size within the valid range.
```
Comment (palisadoes) [inline tests/setup/setImageUploadSize.spec.ts:20]: Why have you removed these clarifying comments?

File: tests/setup/setImageUploadSize.spec.ts [hunk 4]
```diff
@@ -1,63 +1,92 @@
...
   setImageUploadSize,
   validateImageFileSize,
 } from "../../src/setup/setImageUploadSize";
+import fs from "fs";
+import dotenv from "dotenv";
 import { MAXIMUM_IMAGE_SIZE_LIMIT_KB } from "../../src/constants";
 
-/*
-  Test Case 1:
-  Description: setImageUploadSize updates .env_test for the test environment with the correct size.
```
Comment (palisadoes) [inline tests/setup/setImageUploadSize.spec.ts:12]: Why have you removed these clarifying comments?

File: tests/setup/setImageUploadSize.spec.ts [hunk 5]
```diff
@@ -1,63 +1,92 @@
...
-  Description: The validateImageFileSize function returns true for a size within the valid range.
-  Expected Behavior: When validateImageFileSize function is called with a size value within the valid range, it should return true indicating that the size is valid.
+  it("should set the image upload size correctly in .env file", async () => {
+    vi.spyOn(dotenv, "parse").mockResolvedValue({
+      IMAGE_SIZE_LIMIT_KB: "500",
+    });
+    vi.spyOn(fs, "readFileSync").mockReturnValue("IMAGE_SIZE_LIMIT_KB=500");
 
-  Test Case 4:
-  Description: The validateImageFileSize function returns false for a size exceeding the valid range.
```
Comment (palisadoes) [inline tests/setup/setImageUploadSize.spec.ts:24]: Why have you removed these clarifying comments?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by Cioppolo14** [comment]
@aarishshahmohsin Please fix the failed tests.

**Comment by aarishshahmohsin** [comment]
@Cioppolo14 @SiddheshKukade @EshaanAgg @palisadoes I am not able to replicate the test error locally. My tests are passing locally.

**Comment by aarishshahmohsin** [comment]
@palisadoes please review this PR.

**Comment by palisadoes** [comment]
You didn't answer my questions in the previous review. You marked then as resolved without a response

**Comment by aarishshahmohsin** [comment]
> You didn't answer my questions in the previous review. You marked then as resolved without a response

I added those comments.

**Comment by aarishshahmohsin** [comment]
@palisadoes Please review this PR. I am not able to work on other issues.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**